app/main.py

```diff
@@ -2,0 +1,41 @@
+class Animal:
+    alive = []
+
+    def __init__(
+            self, name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+
+        Animal.alive.append(self)
+
+    def take_damage(self) -> None:
+        self.health -= 50
+        if self.health <= 0:
+            Animal.alive.remove(self)
```

Comment:
In the `take_damage` method, before removing `self` from `Animal.alive`, you should check if `self` is actually in the list to prevent potential errors. Consider using a condition like `if self in Animal.alive:` before attempting to remove it.